pyorbital/sno_utils.py

```diff
@@ -0,0 +1,194 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+
+# Copyright (c) 2019 Pytroll
+
+# Author(s):
+
+#   Adam.Dybbroe <dam.dybbroe@smhi.se>
+
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+"""Utility functions to find simultaneous nadir overpasses (SNOs)."""
+
+import os
+from datetime import timedelta
+from pyorbital.tlefile import Tle
+from pyresample.spherical_geometry import Coordinate, Arc
+import math
+import logging
+import yaml
+try:
+    from yaml import UnsafeLoader
+except ImportError:
+    from yaml import Loader as UnsafeLoader
+try:
+    # python 3.3+
+    from collections.abc import Mapping
+except ImportError:
+    # deprecated (above can't be done in 2.7)
+    from collections import Mapping
+
+LOG = logging.getLogger(__name__)
+
+TLE_BUFFER = {}
+TLE_SATNAME = {'Suomi-NPP': 'SUOMI NPP',
+               'NOAA-20': 'NOAA 20',
+               'EOS-Aqua': 'AQUA',
+               'Metop-B': 'METOP-B',
+               'NOAA-19': 'NOAA 19',
+               'NOAA-18': 'NOAA 18'}
+
+
+class NoTleFile(Exception):
+    """Exception to catch missing TLE file."""
+
+    pass
+
+
+def _recursive_dict_update(d, u):
+    """Recursive dictionary update.
+
+    Copied from:
+
+        http://stackoverflow.com/questions/3232943/update-value-of-a-nested-dictionary-of-varying-depth
+
+    """
+    for k, v in u.items():
+        if isinstance(v, Mapping):
+            r = _recursive_dict_update(d.get(k, {}), v)
+            d[k] = r
+        else:
+            d[k] = u[k]
+    return d
+
+
+def get_config(configfile):
+    """Get the configuration from file."""
+    config = {}
+    with open(configfile, 'r') as fp_:
+        config = _recursive_dict_update(config, yaml.load(fp_, Loader=UnsafeLoader))
```

Comment:
Is `UnsafeLoader` needed because of loading Python objects, or would `SafeLoader`/`BaseLoader` work?

Reply:
Oh yes, you are right, as for now the SafeLoader is okay. Thx @pnuu

Reply:
Fixed!